src/Aspire.Hosting.DevTunnels/api/Aspire.Hosting.DevTunnels.cs

```diff
@@ -0,0 +1,74 @@
+
+        public static ApplicationModel.IResourceBuilder<DevTunnels.DevTunnelResource> WithReference(this ApplicationModel.IResourceBuilder<DevTunnels.DevTunnelResource> tunnelBuilder, ApplicationModel.EndpointReference targetEndpoint, bool allowAnonymous) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<DevTunnels.DevTunnelResource> WithReference(this ApplicationModel.IResourceBuilder<DevTunnels.DevTunnelResource> tunnelBuilder, ApplicationModel.EndpointReference targetEndpoint) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<TResource> WithReference<TResource>(this ApplicationModel.IResourceBuilder<TResource> builder, ApplicationModel.IResourceBuilder<ApplicationModel.IResourceWithEndpoints> targetResource, ApplicationModel.IResourceBuilder<DevTunnels.DevTunnelResource> tunnelResource)
+            where TResource : ApplicationModel.IResourceWithEnvironment { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<DevTunnels.DevTunnelResource> WithReference<TResource>(this ApplicationModel.IResourceBuilder<DevTunnels.DevTunnelResource> tunnelBuilder, ApplicationModel.IResourceBuilder<TResource> resourceBuilder, DevTunnels.DevTunnelPortOptions? portOptions = null)
+            where TResource : ApplicationModel.IResourceWithEndpoints { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<DevTunnels.DevTunnelResource> WithReference<TResource>(this ApplicationModel.IResourceBuilder<DevTunnels.DevTunnelResource> tunnelBuilder, ApplicationModel.IResourceBuilder<TResource> resourceBuilder, bool allowAnonymous)
+            where TResource : ApplicationModel.IResourceWithEndpoints { throw null; }
+    }
+}
+
+namespace Aspire.Hosting.DevTunnels
+{
+    public sealed partial class DevTunnelOptions
+    {
+        public bool AllowAnonymous { get { throw null; } set { } }
+
+        public string? Description { get { throw null; } set { } }
+
+        public System.Collections.Generic.List<string>? Labels { get { throw null; } set { } }
+    }
+
+    public sealed partial class DevTunnelPortOptions
+    {
+        public bool? AllowAnonymous { get { throw null; } set { } }
+
+        public string? Description { get { throw null; } set { } }
+
+        public System.Collections.Generic.List<string>? Labels { get { throw null; } set { } }
+
+        public string? Protocol { get { throw null; } set { } }
+    }
+
+    public sealed partial class DevTunnelPortResource : ApplicationModel.Resource, IResourceWithServiceDiscovery, ApplicationModel.IResourceWithEndpoints, ApplicationModel.IResource, ApplicationModel.IResourceWithWaitSupport
+    {
+        public const string TunnelEndpointName = "tunnel";
+        public DevTunnelPortResource(string name, DevTunnelResource tunnel, ApplicationModel.EndpointReference targetEndpoint, DevTunnelPortOptions? options = null) : base(default!) { }
+
+        public DevTunnelResource DevTunnel { get { throw null; } }
```

Comment:
Should this be modeled as a `IResourceWithParent` relationship?

cc @DamianEdwards

Reply:
No, as that adds behavior we don't want in this case (i.e. that the child resource inherits all its status from the parent) as dev tunnel port resources are custom resources that manage their own lifetime by handling events on the parent resource as well as the resource the target endpoint is on.